main/src/ca/uwaterloo/flix/tools/pkg/MavenPackageManager.scala

```diff
@@ -31,27 +36,44 @@ object MavenPackageManager {
     * Installs all MavenDependencies for a Manifest,
     * including transitive dependencies using coursier.
     */
-  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[Unit, PackageError] = {
+  def installAll(manifest: Manifest)(implicit out: PrintStream): Result[List[Path], PackageError] = {
     val depStrings = getMavenDependencyStrings(manifest)
 
+    val cacheString = "./lib/cache"
+    Files.createDirectories(Paths.get(cacheString))
+
     Result.sequence(depStrings.map(createCoursierDependencies)).flatMap { deps =>
-      try {
+      val l = try {
+        val cache: Cache[Task] = FileCache()
+          .withLocation(cacheString)
+
         val res = deps.foldLeft(Resolve())((res, dep) => res.addDependencies(dep))
-        val resolution = res.run()
+        val resolution = res.withCache(cache).run()
 
-        val fetch = resolution.dependencies.foldLeft(Fetch())((f, dep) => {
-          out.println(s"Installing ${dep.module.toString()}");
-          f.addDependencies(dep)
+        val resList: collection.mutable.ListBuffer[Path] = collection.mutable.ListBuffer.empty
+        val fetch = resolution.dependencies.foldLeft(Fetch())(
+          (f, dep) => {
+            val moduleName = dep.module.toString()
+            val moduleNamePath = moduleName.replaceAll("[^a-zA-Z0-9-]", "/")
+            val versionString = dep.version
+            val fileName = s"${moduleNamePath.split('/').last}-$versionString.jar"
+            val filePrefix = "https/repo1.maven.org/maven2"
+            val path = Paths.get(s"$cacheString/$filePrefix/$moduleNamePath/$versionString/$fileName")
```

Comment:
We have to be very careful with paths.

In essence we should never use `/` or `\` ourselves. What we want is to call `Path.resolve`.